cdap-app-templates/cdap-etl/cdap-etl-proto/src/main/java/io/cdap/cdap/etl/proto/connection/ConnectorDetail.java

```diff
@@ -27,15 +30,29 @@
  */
 public class ConnectorDetail {
   private final Set<PluginDetail> relatedPlugins;
+  private final Set<SampleType> supportedSampleTypes;
 
-  public ConnectorDetail(Set<PluginDetail> relatedPlugins) {
+  public ConnectorDetail(Set<PluginDetail> relatedPlugins, Set<SampleType> supportedSampleTypes) {
     this.relatedPlugins = relatedPlugins;
+    if (supportedSampleTypes != null) {
```

Comment:
If you expect it to be null (which cases?) you should mark it nullable/

Reply:
It's expected to be null e.g. in the call from the other constructor. I've marked it `@Nullable`, thanks!